packages/rich-text/src/plugins/shared/MarkPlugin.js

```diff
@@ -1,17 +1,16 @@
 import isHotkey from 'is-hotkey';
 import markDecorator from './MarkDecorator';
 import { haveMarks } from './UtilHave';
-import { css, cx } from 'emotion';
+import { css } from 'emotion';
 import tokens from '@contentful/forma-36-tokens';
 
 const styles = {
+  // TODO: Ensure a bold mark on a headline makes the heading appear even
+  //  bolder. Consider to make headings non-bold by default.
   bold: css({
     color: 'inherit',
     fontWeight: tokens.fontWeightDemiBold,
   }),
-  headingBold: css({
-    fontWeight: 900, // Make headings which are already bold even bolder.
-  }),
```

Comment:
And to be clear, this is the "broken" logic? i.e., this never really made the headings bolder? (hence why we can get away with no more RegExp parsing)

Reply:
It did at some point (I think I originally implemented it and have tested it for sure) but I guess in some slate upgrade or refactoring it just didn't work anymore.